src/SocketRegistry.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import {IRegistryCoordinator} from "./interfaces/IRegistryCoordinator.sol";
+import {ISocketRegistry} from "./interfaces/ISocketRegistry.sol";
+
+/**
+ * @title A `Registry` that keeps track of operator sockets.
+ * @author Layr Labs, Inc.
+ */
+contract SocketRegistry is ISocketRegistry {
+
+    /// @notice The address of the RegistryCoordinator
+    address public immutable registryCoordinator;
+
+    /// @notice A mapping from operator IDs to their sockets
+    mapping(bytes32 => string) public operatorIdToSocket;
+
+    /// @notice A modifier that only allows the RegistryCoordinator to call a function
+    modifier onlyRegistryCoordinator() {
+        require(msg.sender == address(registryCoordinator), "SocketRegistry.onlyRegistryCoordinator: caller is not the RegistryCoordinator");
+        _;
+    }
+
+    /// @notice A modifier that only allows the owner of the RegistryCoordinator to call a function
+    modifier onlyCoordinatorOwner() {
+        require(msg.sender == IRegistryCoordinator(registryCoordinator).owner(), "SocketRegistry.onlyCoordinatorOwner: caller is not the owner of the registryCoordinator");
+        _;
+    }
+
+    constructor(IRegistryCoordinator _registryCoordinator) {
+        registryCoordinator = address(_registryCoordinator);
+    }
+
+    /// @notice sets the socket for an operator only callable by the RegistryCoordinator
+    function setOperatorSocket(bytes32 _operatorId, string memory _socket) external onlyRegistryCoordinator {
+        operatorIdToSocket[_operatorId] = _socket;
+    }
+
+    /// @notice migrates the sockets for a list of operators only callable by the owner of the RegistryCoordinator
+    function migrateOperatorSockets(bytes32[] memory _operatorIds, string[] memory _sockets) external onlyCoordinatorOwner {
```

Comment:
Is it okay if this won't emit an event in the registry coord?

Reply:
yes the events are going to eventually be deprecated

Reply:
We should keep the events around so we know when they change?

Reply:
Ya good point we can leave those but we wont need it in the migration function